lib/delivery-locations-resolver.js

```diff
@@ -311,10 +304,34 @@ class DeliveryLocationsResolver {
           }
           // Establish default for Electronic Document Delivery flag:
           item.eddRequestable = !!item.eddRequestable
-          item.deliveryLocation = this.formatLocations(item.deliveryLocation, deliveryLocationTypes)
+          const deliveryLocationsWithFilteredScholarRooms = this.filterByTypeAndScholarLocations(item.deliveryLocation, deliveryLocationTypes, scholarRoom)
+          item.deliveryLocation = this.formatLocations(deliveryLocationsWithFilteredScholarRooms)
           return item
         })
       })
   }
+  static filterByTypeAndScholarLocations (locations, deliveryLocationTypes, scholarRoom) {
+    if (!locations || !locations.length) return []
+    const onlyDeliverabletToLpaOrSchomburg = locations.filter((location) => {
+      return location.code.startsWith('ma')
+    }).length === 0
+    // remove scholar locations that are not the scholar room, if a specific
+    // scholar room exists.s
+    // Filter out anything not matching the specified deliveryLocationType
+    return locations.filter((location) => {
+      if (deliveryLocationTypes) {
+        return location.deliveryLocationTypes &&
+          arrayIntersection(location.deliveryLocationTypes, deliveryLocationTypes).length > 0
+      } else return true
+    })
+      .filter((location) => {
+        const locationIsNotScholarRoom = !location.deliveryLocationTypes.includes('Scholar')
+        // If an item is not deliverable to SASB, don't return scholar rooms
+        if (onlyDeliverabletToLpaOrSchomburg) {
+          return locationIsNotScholarRoom
+        } else return locationIsNotScholarRoom || (scholarRoom && location.code === scholarRoom.code)
+      })
+  }
```

Comment:
I think the `onlyDeliverabletToLpaOrSchomburg` is there to handle a data issue, but we're not sure the data issue exists? It seems like the issue it's addressing is the case where the set of unfiltered delivery locations includes only LPA/SC locations plus some SASB scholar rooms. But is there any known holding location that produces that scenario? I don't know of any and I think we would want to fix it as a data issue if we did find such a case. Unless this addresses known issues, I'd vote to simplify this function to just: 1) filter out any locations with disagreeing `deliveryLocationTypes`, and 2) filter out any 'Scholar' tagged location that does not match `scholarRoom`